bulk/v1/batch.go

```diff
@@ -217,14 +217,26 @@ func (b *Batch) Results(ctx context.Context) (BatchResult, error) {
 		return result, err
 	}
 	var requestRecords []map[string]interface{}
+	requestRecords, err = b.requestRecords(ctx)
+	if err != nil {
+		return result, fmt.Errorf("error retrieving request: %w", err)
+	}
 	for i, record := range records {
+		fields := map[string]interface{}{}
+		if i < len(requestRecords) {
+			fields = requestRecords[i]
+		}
+		jobRecord := bulk.JobRecord{
+			ID: record.ID,
+			UnprocessedRecord: bulk.UnprocessedRecord{
```

Comment:
I wonder if this should just be `Record` (rather than "Unprocessed"). Although I suspect it's the former due to embedding, so maybe not an easy/small change.

Reply:
I'm a little unsure about this comment. 

Are you saying:

```
		jobRecord := bulk.JobRecord{
			ID: record.ID,
		}
		if record.Success {
			jobRecord.Fields = fields
		} else {
			jobRecord.UnprocessedRecord.Fields = fields
                }
```

Aren't they equivalent? Since `Fields` can be accessed either directly on `JobRecord` or through `JobRecord.UnprocessedRecord`?

```
type JobRecord struct {
	ID string
	UnprocessedRecord
}
```

Reply:
I was saying that `UnprocessedRecord` is sort of a misnomer; it's really `Record`. So you could image the `JobRecord` type looking like:

```go
type JobRecord struct {
	ID string
	Record UnprocessedRecord
}
```

At which point you say, why not just rename `UnprocessedRecord` to `Record`.

More of an observation/musing than a request for this PR.